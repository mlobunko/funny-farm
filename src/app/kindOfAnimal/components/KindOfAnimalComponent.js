import React from 'react';
import PropTypes from 'prop-types';

import { StyledAnimation, StyledKind } from './styles';

export const KindOfAnimalComponent = ({ clickedAnimal }) => (
  <React.Fragment>
    {clickedAnimal ? (
      <StyledKind>{clickedAnimal}</StyledKind>
    ) : (
      <StyledAnimation>Touch The Animal</StyledAnimation>
    )}
  </React.Fragment>
);

export default KindOfAnimalComponent;

KindOfAnimalComponent.proptypes = {
  clickedAnimal: PropTypes.string.isRequired
};
