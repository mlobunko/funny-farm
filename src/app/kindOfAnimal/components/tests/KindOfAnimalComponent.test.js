import React from 'react';
import { shallow } from 'enzyme';

import { KindOfAnimalComponent } from '../KindOfAnimalComponent';

describe('render', () => {
  it('renders StyledKind', () => {
    const clickedAnimal = 'pig';
    const wrapper = shallow(
      <KindOfAnimalComponent clickedAnimal={clickedAnimal} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders StyledAnimation', () => {
    const wrapper = shallow(<KindOfAnimalComponent />);
    expect(wrapper).toMatchSnapshot();
  });
});
