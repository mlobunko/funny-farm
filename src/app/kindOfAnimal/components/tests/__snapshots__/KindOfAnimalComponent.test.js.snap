// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders StyledAnimation 1`] = `
<Fragment>
  <styled.div>
    Touch The Animal
  </styled.div>
</Fragment>
`;

exports[`render renders StyledKind 1`] = `
<Fragment>
  <styled.div>
    pig
  </styled.div>
</Fragment>
`;
